src/host/settings.cpp

```diff
@@ -326,6 +326,22 @@ void Settings::Validate()
     WI_ClearAllFlags(_wFillAttribute, ~(FG_ATTRS | BG_ATTRS));
     WI_ClearAllFlags(_wPopupFillAttribute, ~(FG_ATTRS | BG_ATTRS));
 
+    // If the extended color options are set to invalid values (all the same color), reset them.
+    if (_CursorColor != Cursor::s_InvertCursorColor && _CursorColor == _DefaultBackground)
+    {
+        _CursorColor = Cursor::s_InvertCursorColor;
+    }
+
+    if (_DefaultForeground != INVALID_COLOR && _DefaultForeground == _DefaultBackground)
+    {
+        _DefaultForeground = _DefaultBackground = INVALID_COLOR;

```

Comment:
Sorry, does something later in the startup process find this and fix it? That's not totally clear.

Reply:
Oh, yeah. Whenever Settings is asked what the default color is it checks for INVALID_COLOR and returns [either the indexed color or the color]. `INVALID_COLOR` is just the value of "not set, use the indexed colors".

Reply:
That's here: 

https://github.com/microsoft/terminal/blob/65a6d08d87a3bdf4140d0d661bee83131e178c79/src/host/settings.cpp#L869-L881

Reply:
(and later in the file.) It was done this way presumably to support the terminal colors being toggled on/off at runtime.

Ideally it'd be an std::optional COLORREF, but it wasn't done that way.

Reply:
Leave comment plx.